src/compression.rs

```diff
@@ -1497,17 +1550,18 @@ pub fn compression_step(
 
     // build the egraph. We'll just be using this as a structural hasher we don't use rewrites at all. All eclasses will always only have one node.
     let mut egraph: EGraph = Default::default();
-    let programs_node = egraph.add_expr(programs_expr.into());
+    let train_programs_node = egraph.add_expr(train_programs_expr.into());
+    let test_programs_node = if let Some(e) = test_programs_expr { Some(egraph.add_expr(e.into())) } else { None };
```

Comment:
So, I guess inserting the test programs into the egraph but then getting the roots and treenodes from the training nodes alone makes the most sense?

Also, see comment for line 96 in `compress.rs`. Squawk or whatever it is that parrots say.

Reply:
egraph stuff: sure thats fine

line 96: same soln as before squawk